### ISSUE TITLE
Docs on dotnet test arguments coming from vstest are missing

### ISSUE THREAD

**Issue by dhrumil0902** [dotnet/docs#41773]
### Proposed topic or title

Vstest Arguments Transferred to DotNet Test

### Reason for the article

According to the dotnet vstest [documentation page](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-vstest), the command is superseded by the dotnet test command, which can now be used to run assemblies.

Which, I believe, was done as a result of https://github.com/microsoft/vstest/issues/1453 request in a scope of https://github.com/microsoft/vstest-docs/pull/196.

dotnet vstest is able to run tests from the assemblies in parallel if --parallel flag is provided. Per the referenced https://github.com/microsoft/vstest-docs/pull/196, there was an intention to support the same behavior for the dotnet test command as well.

Success criteria (as per user ask)

The "dotnet test" command can be used in place of the "dotnet vstest" command.
The "dotnet test" command can accept a list of dlls
Using "dotnet test" and targeting a list of dlls, there is an argument for "--parallel" which will run the tests from the dlls in parallel.
Using "dotnet test" and targeting a dll, the inclusion or exclusion of the "--no-build" argument will have no impact on the behavior of execution.
My local testing confirms that --parallel flag is respected and tests are run in parallel; test from the assemblies are run sequentially otherwise. Note, that I'm talking parallel execution of the assemblies themselves and not the internal assembly-level of tests parallelization, which is handled by NUnit and its adapter for me.

However, there is no documentation for that feature at the dotnet test page and in the output of the dotnet test --help command, which is confusing.

Is parallel execution of tests from assemblies supported by the dotnet test command officially, and it's safe to rely on that, or it's an implementation detail exposed accidentally?
Should the documentation for dotnet test be updated to reflect that?

### Article abstract

More information about how exisiting features of vstest can be used in dotnet test.

### Relevant searches

dotnet test --parallel
dotnet test <file>
dotnet test and vstest integration

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 61ca4064-7481-a263-60fc-c70127f2ad16
* Version Independent ID: 2878af44-8208-7bd8-46d4-3acda3e82ce0
* Platform ID: 3cb4c6af-8b0f-5900-031d-2b8259c5f388
* Content: [dotnet test command - .NET CLI](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-test?tabs=dotnet-test-with-vstest)
* Content Source: [docs/core/tools/dotnet-test.md](https://github.com/dotnet/docs/blob/main/docs/core/tools/dotnet-test.md)
* Service: **dotnet-fundamentals**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

**Comment by meaghanlewis** [comment]
Similarly to #35931, we don't plan to add any additional options that aren't officially specified and documented in the SDK. Not every argument from vstest was transferred over.

By default, the `dotnet test` command runs test assemblies in parallel.

You can disable parallel execution with: `dotnet test -p:TestTfmsInParallel=false` or use the `-maxcpucount` switch to set a limit for the number of parallel processes to run.

This is described in: 
- https://learn.microsoft.com/dotnet/core/tools/dotnet-test-vstest#description
- https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/sdk#run-tests-in-parallel
